core_types/src/ugly.rs

```diff
@@ -0,0 +1,23 @@
+use std::fmt::Debug;
```

Comment:
I'm not very happy with `ugly` in core_types.  This seems to mean that we are explicitly taking on some tech-debt, and if so I'd like to see that justified and documented in the PR under followups, and even better in the tech-debt section of the tree.

Reply:
Actually I think now that these printlns are removed, we can do away with `lax_send`. This was part of the path of discovery. I think in the current approach, it's actually acceptable to send on a channel and ignore whether it succeeded or not, because in each of these cases, that's now handled by the corresponding Future. I'll move these functions elsewhere.

I still like the idea of an `ugly` module, because if we're going to have tech-debt, I want it to be highly visible and slightly uncomfortable. But after this change, it would be empty.

Reply:
@zippy where would be a good place to move `lax_send_sync`? I want to keep the function but move it out of `ugly`